base_layer/core/src/base_node/proto/chain_metadata.rs

```diff
@@ -40,13 +40,19 @@ impl TryFrom<proto::ChainMetadata> for ChainMetadata {
         let mut acc_diff = [0; ACC_DIFFICULTY_ARRAY_LEN];
         acc_diff.copy_from_slice(&metadata.accumulated_difficulty[0..ACC_DIFFICULTY_ARRAY_LEN]);
         let accumulated_difficulty = u128::from_be_bytes(acc_diff);
+        let height_of_longest_chain = metadata
+            .height_of_longest_chain
+            .ok_or_else(|| "Height of longest chain is missing".to_string())?;
+        let pruning_horizon = if metadata.effective_pruned_height == 0 {
+            metadata.effective_pruned_height
+        } else {
+            height_of_longest_chain - metadata.effective_pruned_height
```

Comment:
This is untrusted data, so use `saturating_sub` here (`Sub` impl will panic on underflow)